js/src/javascript/beautifier.js

```diff
@@ -691,10 +691,10 @@ Beautifier.prototype.handle_start_block = function(current_token) {
     )) {
     // We don't support TypeScript,but we didn't break it for a very long time.
     // We'll try to keep not breaking it.
-    if (!in_array(this._last_last_text, ['class', 'interface'])) {
-      this.set_mode(MODE.ObjectLiteral);
-    } else {
+    if (in_array(this._last_last_text, ['class', 'interface']) && !!second_token.whitespace_before) {
```

Comment:
I don't think whitespace is a good indicator for what you're trying to do here. 

The following two inputs should produce the same output, but I don't think they do with this change:
* `var a={class:{a:1}}`
* `var a = { class : { a : 1 } }`

Reply:
@bitwiseman Both the cases you have stated resulted in the same output using whitespace however I took your comments into consideration and changed the condition to look for the text for the `second_token`.